src/main/java/software/amazon/profiler/BasePlugin.java

```diff
@@ -38,9 +40,15 @@ public class BasePlugin {
     // One profiler per JVM process
     transient volatile Profiler _profiler;
 
-    public synchronized void startProfiler(String profilingGroupName, boolean heapSummaryEnabled) {
+    private final static Random random = ThreadLocalRandom.current();
+
+    public synchronized void startProfiler(String profilingGroupName, boolean heapSummaryEnabled, double probability) {
         Profiler profiler = _profiler;
         if (profiler == null) {
+            if (random.nextDouble() > probability) {
```

Comment:
Assuming that the condition should be always true if probability = 0, and be always false if probability = 1, shall we use `>=` instead of `>`?

[nextDouble](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html#nextDouble--)

> Returns a pseudorandom double value between zero (inclusive) and one (exclusive).

Reply:
Yep, good catch. It's a 1/2^54 chance but still.